torch/csrc/jit/tensorexpr/external_functions.cpp

```diff
@@ -196,6 +204,29 @@ void nnc_aten_mean(
   }
 }
 
+void nnc_aten_addmm(
+    int64_t bufs_num,
+    void** buf_data,
+    int64_t* buf_ranks,
+    int64_t* buf_dims,
+    int8_t* buf_dtypes,
+    int64_t args_num,
+    int64_t* extra_args) {
+  std::vector<at::Tensor> tensors =
+      constructTensors(bufs_num, buf_data, buf_ranks, buf_dims, buf_dtypes);
+
+  at::Tensor& r = tensors[0];
+  const at::Tensor& x = tensors[1];
+  const at::Tensor& y = tensors[2];
+  const at::Tensor& z = tensors[3];
+  try {
+    at::addmm_out(r, x, y, z, extra_args[0], extra_args[1]);
+  } catch (...) {
+  }
+}
+
+#ifndef C10_MOBILE
```

Comment:
Just curious, why are these registrations not needed for mobile?

Reply:
These registrations are used by the compiler at AOT compilation time. The registrations are not needed by the runtime as the assembly code will directly call the external functions without going through the registration table.

To include these we will need add a couple more source files to the runtime build, so I simply bypass them here.

C10_MOBILE implies mobile + runtime (no AOT / JIT compilation). Maybe we can create a dedicated macro for the runtime part (I'm little hesitate as new macros will add maintenance complexity).

Reply:
These registrations are needed for llvm codegen to figure out what function to call to when it sees a "nnc_aten_conv2d" external call. I assume that codegen would be happening at the server side (since we're doing AOT) where all these registrations are present. Once the code is generated, we no longer need the registrations - thus in the mobile build we don't need them, we only need the functions themselves so that linker would find them.